packages/apify/src/actor.ts

```diff
@@ -512,15 +514,40 @@ export class Actor<Data extends Dictionary = Dictionary> {
      * For more information, see the [Actor Runs](https://docs.apify.com/api/v2#/reference/actor-runs/) API endpoints.
      * @ignore
      */
-    async setStatusMessage(statusMessage: string): Promise<ClientActorRun> {
+    async setStatusMessage(statusMessage: string, options?: SetStatusMessageOptions): Promise<ClientActorRun> {
+        const { isStatusMessageTerminal, level } = options || {};
         ow(statusMessage, ow.string);
+        ow(isStatusMessageTerminal, ow.optional.boolean);
+
+        statusMessage = `[Status message]: ${statusMessage}`;
+
+        switch (level) {
+            case LogLevel.DEBUG:
+                log.debug(statusMessage);
+                break;
+            case LogLevel.WARNING:
+                log.warning(statusMessage);
+                break;
+            case LogLevel.ERROR:
+                log.error(statusMessage);
+                break;
+            default:
+                log.info(statusMessage);
+                break;
+        }
+
+        const storageClient = this.config.getStorageClient();
+        await storageClient.setStatusMessage?.(statusMessage, { isStatusMessageTerminal, level });
 
         const runId = this.config.get('actorRunId')!;
-        if (!runId) {
-            throw new Error(`Environment variable ${ENV_VARS.ACTOR_RUN_ID} is not set!`);
+        if (runId) {
+            const run = await this.apifyClient.run(runId).get();
+            if (run) {
+                return run;
+            }
         }
 
-        return this.apifyClient.run(runId).update({ statusMessage });
+        return null as unknown as ClientActorRun;
```

Comment:
maybe empty object will be better than `null`? at least it wont fail on property access